mllam_data_prep/derived_variables.py

```diff
@@ -0,0 +1,459 @@
+import importlib
+import sys
+
+import numpy as np
+import xarray as xr
+from loguru import logger
+
+
+def derive_variables(fp, derived_variables, chunking):
+    """
+    Load the dataset, and derive the specified variables
+
+    Parameters
+    ---------
+    fp : str
+        Filepath to the source dataset, for example the path to a zarr dataset
+        or a netCDF file (anything that is supported by `xarray.open_dataset` will work)
+    derived_variables : dict
+        Dictionary with the variables to derive
+        with keys as the variable names and values with entries for
+        kwargs and function to use in the calculation
+    chunking: dict
+        Dictionary with keys as the dimensions to chunk along and values
+        with the chunk size
+
+    Returns
+    -------
+    ds : xr.Dataset
+        Dataset with derived variables included
+    """
+    logger.info("Deriving variables")
+
+    try:
+        ds = xr.open_zarr(fp)
+    except ValueError:
+        ds = xr.open_dataset(fp)
+
+    ds_subset = xr.Dataset()
+    ds_subset.attrs.update(ds.attrs)
+    for _, derived_variable in derived_variables.items():
+        required_kwargs = derived_variable.kwargs
```

Comment:
I think it's fine to keep it as is, since I don't think my suggestion makes it simpler, and since the README states clearly what the kwargs section is used to.